core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java

```diff
@@ -1063,4 +1064,26 @@ public void reset() {
       lock.writeLock().unlock();
     }
   }
+
+  public PluginInterface findPluginWithId( Class<? extends PluginTypeInterface> pluginType, String pluginId, boolean waitForPluginToBeAvailable ) {
+    return waitForPluginToBeAvailable && findPluginWithId(  pluginType,  pluginId ) == null
```

Comment:
I'd pull the findPluginWithId call out of the conditional so you don't need to invoke it twice.